// @medusajs/utils/dist/promotion/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromotionRuleOperator = exports.ApplicationMethodAllocation = exports.ApplicationMethodTargetType = exports.ApplicationMethodType = exports.PromotionType = void 0;
var PromotionType;
(function (PromotionType) {
    PromotionType["STANDARD"] = "standard";
    PromotionType["BUYGET"] = "buyget";
})(PromotionType || (exports.PromotionType = PromotionType = {}));
var ApplicationMethodType;
(function (ApplicationMethodType) {
    ApplicationMethodType["FIXED"] = "fixed";
    ApplicationMethodType["PERCENTAGE"] = "percentage";
})(ApplicationMethodType || (exports.ApplicationMethodType = ApplicationMethodType = {}));
var ApplicationMethodTargetType;
(function (ApplicationMethodTargetType) {
    ApplicationMethodTargetType["ORDER"] = "order";
    ApplicationMethodTargetType["SHIPPING"] = "shipping";
    ApplicationMethodTargetType["ITEM"] = "item";
})(ApplicationMethodTargetType || (exports.ApplicationMethodTargetType = ApplicationMethodTargetType = {}));
var ApplicationMethodAllocation;
(function (ApplicationMethodAllocation) {
    ApplicationMethodAllocation["EACH"] = "each";
    ApplicationMethodAllocation["ACROSS"] = "across";
})(ApplicationMethodAllocation || (exports.ApplicationMethodAllocation = ApplicationMethodAllocation = {}));
var PromotionRuleOperator;
(function (PromotionRuleOperator) {
    PromotionRuleOperator["GTE"] = "gte";
    PromotionRuleOperator["LTE"] = "lte";
    PromotionRuleOperator["GT"] = "gt";
    PromotionRuleOperator["LT"] = "lt";
    PromotionRuleOperator["EQ"] = "eq";
    PromotionRuleOperator["NE"] = "ne";
    PromotionRuleOperator["IN"] = "in";
})(PromotionRuleOperator || (exports.PromotionRuleOperator = PromotionRuleOperator = {}));
